titus-api/src/main/java/com/netflix/titus/api/jobmanager/model/job/TaskStatus.java

```diff
@@ -201,20 +219,28 @@ public Builder toBuilder() {
     }
 
     public static class Builder extends AbstractBuilder<TaskState, Builder, TaskStatus> {
+        private List<ContainerState> containerState;
+
         private Builder() {
         }
 
         private Builder(TaskStatus status) {
             super(status);
         }
 
+        public Builder withContainerState(List<ContainerState> containerState) {
+            this.containerState = containerState;
+            return this;
+        }
+
         @Override
         public TaskStatus build() {
             return new TaskStatus(
                     state,
                     reasonCode == null ? TaskStatus.REASON_UNKNOWN : reasonCode,
                     toCompleteReasonMessage(),
-                    timestamp
+                    timestamp,
+                    containerState == CollectionsExt.nonNull(containerState) ? new ArrayList<>() : containerState
```

Comment:
Just pass this value to constructor, and make sure it is not null there?
`containerStates` (plural)?